pyccl/halos/profiles.py

```diff
@@ -147,6 +147,36 @@ def _get_plaw_projected(self, cosmo, a):
         """
         return self.precision_fftlog['plaw_projected']
 
+    def __eq__(self, prof2):
+        """ Return `True` if this profile is equivalent to another."""
+        if type(self) != type(prof2):
+            # catch different profile types
+            return False
+
+        params, params2 = self.__dict__, prof2.__dict__
+
+        if params == params2:
+            # same profiles
+            return True
+
```

Comment:
Humm, I'm confused, it seems the only outcome after this line is for this function to return `False`, right? If so, why the extra checks?

Reply:
Good catch, I meant to return `True` if all the checks have passed. I added an extra if block, which if passes, means that the two profiles are indeed equivalent.